source/OptimaJet.DWKit.StarterApplication/Services/Workflow/WorkflowProductService.cs

```diff
@@ -107,24 +107,15 @@ protected async Task CreateWorkflowProcessInstance(Product product)
                 new CreateInstanceParams(
                     product.ProductDefinition.Workflow.WorkflowScheme,
                     product.Id)
-                    {
-                        IdentityId = product.Project.Owner.WorkflowUserId.Value.ToString()
-                    }
+                {
+                    IdentityId = product.Project.Owner.WorkflowUserId.Value.ToString()
+                }
             );
         }
 
         public async Task ProductProcessChangedAsync(ProductProcessChangedArgs args)
         {
-            // Clear PreExecute entries
-            var emptyItems = ProductTransitionRepository.Get()
-                .Where(pt => pt.WorkflowUserId == null && pt.ProductId == args.ProcessId)
-                .Select(pt => pt.Id).ToList();
-            foreach (var item in emptyItems)
-            {
-                await ProductTransitionRepository.DeleteAsync(item);
-            }
-            // Create PreExecute entries
-            await Runtime.PreExecuteFromCurrentActivityAsync(args.ProcessId);
+            await RecreatePreExecuteEntries(args.ProcessId);
```

Comment:
`ClearPreExecuteEntries should be at the beginning of `ReassignUserTasksForProduct`.  A `CreatePreExecuteEntries` should be at the end of `ReassignUserTasksForProduct`.

Reply:
done